### ISSUE TITLE
Block Bindings: Add support for the Cover block background image

### ISSUE THREAD

**Issue by bacoords** [WordPress/gutenberg#63763]
## What problem does this address?

Right now you can tie the cover block's image to the featured image, but that's as dynamic as you can get. Cover blocks make great containers for synced patterns and it would be great to be able to have block overrides on the cover block's image. The controls are already in the block toolbar, so it shouldn't require too much UI adjustment.

## What is your proposed solution?

Enable block bindings/overrides on the cover block's image ID attribute.

**Comment by bacoords** [comment]
@SantosGuillamot In response to [this comment](https://github.com/WordPress/gutenberg/issues/63018#issuecomment-2242859059), do you imagine there being a future abstraction for block bindings that would work for the cover block to opt into? If so, I agree that's worth waiting for.  Otherwise, it seems that the way the cover block sets a background image is pretty unique and might always be a little bit hard-coded.

**Comment by SantosGuillamot** [comment]
I am not familiar with the cover block code, but while working on adding support for some attributes of the image, they could benefit from future `set_inner_html` or `set_outer_html` methods from the HTML API. I assumed that, to cover the different use cases, the cover block would need those as well. But I might be mistaken and maybe that's not the case.

**Comment by justintadlock** [comment]
Just want to echo my support for this specific block to have bindings. It'll open up a lot of custom things I want to do with category/tag/term archives. 

I can actually kind of fake this to setting the Cover (outside the loop) to use the featured image, but it's really just using the first post's image if it exists:

![author-archive-cover](https://github.com/user-attachments/assets/4ba01c07-1d5d-4d8c-8d99-69faaa995407)

I have bunches and bunches of other use cases too.

**Comment by jptweb** [comment]
Yes I am also investigating if there is possible or a workaround. Cover images seem like such a common feature most websites have; and they are generally on every page so they really benefit from overrides and syncing. So simply using a non synced pattern means I may need to go and and re-save hundreds of pages to update there style. I am trying to see if it could be possible with block variations as well; look forward to hearing any viable solutions as a workaround.

A hacky approach I had was to make one non synced pattern that holds the cover element, and inside that put the synced content like text etc as overrides. This has worked ok but it looks a little weird in the patterns menu and still could lead to issues if I ever needed to change the non synced cover.

What i'll probably end up doing is just creating my own block.. but I then feel i'm reinventing the wheel as the cover block already does cool stuff like serve different size images on different devices etc; allows you to choose focal point etc.

EDIT i'm also realizing this would be a lot tricker than though as things like focal point would be unique for each cover etc. there would be a lot of overrides! I suppose for now I will follow this tutorial to create gutenberg blocks with dynamic content https://kinsta.com/blog/dynamic-blocks/ but once again then i'm recreating the wheel here :(

**Comment by Drivingralle** [comment]
+1 for supporting block binding for the cover block image

**Comment by kubiqsk** [comment]
+1 please

**Comment by thomasnavarro** [comment]
+1 for this feature!

I've tried implementing this via plugin but the core doesn't expose the necessary hooks yet. Would love to see this in 6.9!

**Comment by t-hamano** [comment]
This issue has been punted to 6.9. Let's move this issue to the 6.9 Editor Tasks board and see what's possible in 6.9.

**Comment by t-hamano** [comment]
@ockham @cbravobernal I'd love some feedback on whether there are any technical barriers to adding block binding support for the cover block image,  and whether it's worth the effort for WP 6.9 🙏

**Comment by cbravobernal** [comment]
> [@ockham](https://github.com/ockham)[<img alt="" width="12" height="12" src="chrome-extension://hgomfjikakokcbkjlfgodhklifiplmpg/images/wp-logo.png">](https://profiles.wordpress.org/bernhard-reiter) [@cbravobernal](https://github.com/cbravobernal)[<img alt="" width="12" height="12" src="chrome-extension://hgomfjikakokcbkjlfgodhklifiplmpg/images/wp-logo.png">](https://profiles.wordpress.org/cbravobernal) I'd love some feedback on whether there are any technical barriers to adding block binding support for the cover block image, and whether it's worth the effort for WP 6.9 🙏

We still need to land, test and confirm the new approach this image caption PR. https://github.com/WordPress/gutenberg/pull/71483

Regarding Core, we are landing the technical barriers in these PRs:
 - https://github.com/WordPress/wordpress-develop/pull/9702
 - https://github.com/WordPress/wordpress-develop/pull/9469

Once is done, we can start adding more compatibility to different blocks. But we need some time. We are 6 weeks until the last RC of 6.9, so let's wait for a couple of weeks and see how is going.

**Comment by t-hamano** [comment]
This ticket was featured on today's 6.9 Bug Sclub.

Is this issue ready to be tackled now, or do we need to harden Block Bindings a bit more? If it's the latter, let's punt this issue to 7.0.

**Comment by cbravobernal** [comment]
> This ticket was featured on today's 6.9 Bug Scrub.
> 
> Is this issue ready to be tackled now, or do we need to harden Block Bindings a bit more? If it's the latter, let's punt this issue to 7.0.

The issue is being worked in https://github.com/WordPress/gutenberg/pull/72062

The cover background is working. But the "featured image" option in pattern overrides is not working as expected. I'm afraid it won't be ready for 6.9
